File: problems/0003_longest-substring-without-repeating-characters/index-of/Solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max = 0;
        for (int start = 0, end = 0; end < s.length();) {
            final int idx = indexOf(s, s.charAt(end), start, end++);
            if (idx == -1)
                max = Math.max(max, end - start);
            else
                start = idx + 1;
        }
        return max;
    }
    static int indexOf(final String s, final char c, final int start, final int end) {
        for (int i = start; i < end; ++i)
            if (s.charAt(i) == c)
                return i;
        return -1;
    }
}

File: problems/0003_longest-substring-without-repeating-characters/index-of/Solution.php
<?php
class Solution {
    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $max = 0;
        $start = 0;
        $l = strlen($s);
        for ($end = 0; $end < $l;) {
            $idx = self::indexOf($s, $s[$end], $start, $end);
            ++$end;
            if ($idx === FALSE)
                $max = max($max, $end - $start);
            else
                $start = $idx + 1;
        }
        return $max;
    }
    static function indexOf($s, $c, $start, $end) {
        for ($i = $start; $i < $end; ++$i)
            if ($s[$i] === $c)
                return $i;
        return FALSE;
    }
}
